### ISSUE TITLE
ili2pg Interlis Check TID

### ISSUE THREAD

**Issue by jb202302** [claeis/ili2db#512]
Guten Tag

Wir betreiben eine Abgabeplattform für Daten im SIA405 LKMap 2012 Format. In unserem Workflow werden die Daten zuerst mittels den Interlis Tools gecheckt und anschliessend mit dem ili2pg Tool in eine PG Datenbank importiert. Nun gibt es einen Datensatz, welcher eine ungültige TID besitzt (keine StandardOID, wie im Datenmodell verlangt). Im Datencheck der Interlis Tools wird dieser Fehler nicht registriert. Anschliessend bei Check, welcher im ili2pg integriert wird jedoch schon. Der Fehler sieht folgendermassen aus:

Error: line 1944: SIA405_LKMap.SIA405_LKMap.LKLinie: tid 37A546DB: value <37A546DB> is not a valid OID

Gerne würden wir diesen bereits im ersten Datencheck abgreifen. Leider konnte ich im GitHub Verzeichnis keine Informationen zum Datencheck finden, welcher in ili2pg eingebaut ist. Ist das config File dieses Checks öffentlich? Wie könnte dieser Check nach dem Format der OID in den Interlis Tools Check integriert werden?

Vielen Dank für die Hilfe im Voraus!

**Comment by sjib** [comment]
Der VSA hat in seinen Fachprüfungen einen Check für einen gültige TID:

`        !!@ cid="1021";`
`         !!@ ccat="error";`
`         !!@ cmsg_de="Die TID ist keine gültige OID (STANDARDOID). ";`
`		 !!@ cmsg_fr="La TID n'est pas une OID valable (STANDARDOID). ";`
`         !!@ cattr = "";`
`         MANDATORY CONSTRAINT`
`            IGS_match(IGS_getObjectValue(THIS,"TID"),"^[A-Za-z0-9]{16}$");`

siehe https://www.vsa.ch/models/2020/VSADSSMINI_CHECK_FP_2020_2_d_LV95-20211001.ili

Sie können sich analog ein zusätzliches constraint ili schreiben mit dieser Regeln und das zusammen mit LKMap prüfen.

**Comment by edigonzales** [comment]
> Der VSA hat in seinen Fachprüfungen einen Check für einen gültige TID:
> 
> ` !!@ cid="1021";` ` !!@ ccat="error";` ` !!@ cmsg_de="Die TID ist keine gültige OID (STANDARDOID). ";` ` !!@ cmsg_fr="La TID n'est pas une OID valable (STANDARDOID). ";` ` !!@ cattr = "";` ` MANDATORY CONSTRAINT` ` IGS_match(IGS_getObjectValue(THIS,"TID"),"^[A-Za-z0-9]{16}$");`
> 
> siehe https://www.vsa.ch/models/2020/VSADSSMINI_CHECK_FP_2020_2_d_LV95-20211001.ili
> 
> Sie können sich analog ein zusätzliches constraint ili schreiben mit dieser Regeln und das zusammen mit LKMap prüfen.

Und die Funktionen sind in lokal installierten Interlis Tools vorhanden (IGS_macht, IGS_getObjectValue)? Quasi Interlis Tools Standardbibliothek?

**Comment by sjib** [comment]
> Und die Funktionen sind in lokal installierten Interlis Tools vorhanden (IGS_macht, IGS_getObjectValue)? Quasi Interlis Tools Standardbibliothek?

ja, genau. Es ist ein CONTRACTED MODEL das nur mit den INTERLIS Tools läuft:

`INTERLIS 2.3;`

`CONTRACTED MODEL VSADSSMINI_2020_LV95_CHECK_FP (de) AT "http://www.vsa.ch/models" VERSION "2021-10-01" =`
   
`   IMPORTS UNQUALIFIED INTERLIS;`
`   IMPORTS UNQUALIFIED IGSFunction;`
`   IMPORTS UNQUALIFIED MINIFunction;`
`   IMPORTS VSADSSMINI_2020_LV95;`

https://www.vsa.ch/models/2020/IGSFunction.ili

**Comment by olivergrimm** [comment]
@jb202302 Der ilivalidator gibts unter ilicop.ch auch als online-Variante. Die Transferdatei wird nur während der Validierung persistiert und gleich anschliessend gelöscht. Aktuell validieren wir mit ilivalidator 1.13.2 (https://ilicop.ch/api/v1/settings)
